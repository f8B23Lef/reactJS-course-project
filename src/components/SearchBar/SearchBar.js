import React, { useState } from 'react';
import { KEYS } from '../../utils/constants';
import './SearchBar.scss';

export default function SearchBar() {
  const [query, setQuery] = useState('');

  const handleChange = (event) => {
    setQuery(event.target.value);
  };

  const searchQuery = () => {
  };

  const handleKeyDown = (event) => {
    if (event.key === KEYS.ENTER_KEY) {
      searchQuery();
    }
  };

  return (
    <div className='search-bar'>
      <input
        type='text'
        placeholder='What do you want to watch?'
        value={query}
        onChange={handleChange}
        onKeyDown={handleKeyDown}
      />
      <button type='button' onClick={searchQuery}>Search</button>
    </div>
  );
}
