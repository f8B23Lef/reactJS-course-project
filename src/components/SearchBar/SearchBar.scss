@import '../../styles/index.scss';

.search-bar {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 100px 35px;

  input {
    margin-right: 10px;
    padding: 15px 10px;
    width: 500px;
    border-radius: 2px;
    border: none;
    background-color: $translucent-grey;
    color: $white;
    font-size: 14px;
  }

  button {
    padding: 15px 20px;
    background-color: $red;
    border: none;
    border-radius: 4px;
    color: $white;
    cursor: pointer;

    &:hover {
      background-color: $dark-red;
    }
  }
}
