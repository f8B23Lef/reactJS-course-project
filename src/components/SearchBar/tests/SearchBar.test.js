import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { screen, render, fireEvent } from '@testing-library/react';
import SearchBar from '../SearchBar';

const mockedUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
  useLocation: jest.fn().mockImplementation(() => ({ search: '?genre=crime' })),
}));

describe('SearchBar', () => {
  it('renders SearchBar snapshot', () => {
    const { asFragment } = render(<BrowserRouter><SearchBar /></BrowserRouter>);

    expect(asFragment(<SearchBar />)).toMatchSnapshot();
  });

  it('finds placeholder text', () => {
    render(<BrowserRouter><SearchBar /></BrowserRouter>);

    expect(screen.getByPlaceholderText('What do you want to watch?'))
      .toBeInTheDocument();
  });

  it('finds Search button', () => {
    render(<BrowserRouter><SearchBar /></BrowserRouter>);

    expect(screen.getByText('Search')).toBeInTheDocument();
  });

  it('changes value when types', () => {
    render(<BrowserRouter><SearchBar /></BrowserRouter>);

    const input = screen.getByPlaceholderText('What do you want to watch?');
    fireEvent.change(input, { target: { value: 'movie1' } });

    expect(input).toHaveValue('movie1');
  });

  it('calls navigate() on Search button click', () => {
    render(<BrowserRouter><SearchBar /></BrowserRouter>);

    const input = screen.getByPlaceholderText('What do you want to watch?');
    fireEvent.change(input, { target: { value: 'movie1' } });
    const button = screen.getByText('Search');
    fireEvent.click(button);

    expect(mockedUseNavigate).toHaveBeenCalledWith({
      pathname: '/search/movie1',
      search: '?genre=crime',
    });
  });

  it('calls navigate() on Enter key click', () => {
    render(<BrowserRouter><SearchBar /></BrowserRouter>);

    const input = screen.getByPlaceholderText('What do you want to watch?');
    fireEvent.change(input, { target: { value: 'movie1' } });
    fireEvent.keyDown(input, { key: 'Enter' });

    expect(mockedUseNavigate).toHaveBeenCalledWith({
      pathname: '/search/movie1',
      search: '?genre=crime',
    });
  });
});
