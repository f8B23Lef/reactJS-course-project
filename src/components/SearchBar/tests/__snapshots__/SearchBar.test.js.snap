// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders SearchBar snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-bar"
  >
    <input
      placeholder="What do you want to watch?"
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
